### PR TITLE
SW-3877: Added prettier plugin for sorting imports

### PR DESCRIPTION
**Pull request by tommylau523**
The PR made changes to almost every file as part of reformatting. Please take a look at `.prettierrc`, `.prettierignore`, and `package.json`.

The imports are arranged via the following order:

- react packages
- third-party packages
- third-party packages initialed with \@
- paths initialized with src/
- relative paths

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -4,5 +4,9 @@
   "singleQuote": true,
   "jsxSingleQuote": true,
   "tabWidth": 2,
-  "printWidth": 120
+  "printWidth": 120,
+  "plugins": ["@trivago/prettier-plugin-sort-imports"],
+  "importOrder": ["^react(.*)", "<THIRD_PARTY_MODULES>", "@/(.*)", "^src/(.*)$", "^[./]"],
```
Comment (nickgraz) [inline .prettierrc:9]: This is great! I've been doing this manually for way too long

**Comment by tommylau523** [comment]
I would also like some advice on the preferred conventions on making sweeping formatting changes, and how to split them up into different commits of ease of review. @nickgraz

**Comment by nickgraz** [comment]
I haven't been here long enough to see something like this roll through, so I'm not sure if we have any conventions around these kinds of sweeping changes. I think since this one is only dealing with imports, I think it is alright to have one giant PR.

If you really want to split it up, doing each subdirectory under `src` in its own PR would be sufficient. `src/services`, `src/components`, etc...

**Comment by tommylau523** [comment]
I needed to roll back the import changes to avoid merge conflicts. I plan to merge this first, and create another PR for merging changes to import statements wholesale. 

Future changes and commit can run `yarn format` to avoid merge conflicts

**Comment by nickgraz** [comment]
Unfortunately, we have prettier check in GH actions, so I think you have to install it and do all the formatting in the same PR.

**Comment by alexTerraformation** [comment]
> Unfortunately, we have prettier check in GH actions, so I think you have to install it and do all the formatting in the same PR.

This feels like a reasonable protection bypass candidate

File: .prettierrc (hunk 1 above)
Comment (joshbuchea) [reply .prettierrc:9]: +1 this is great, thank you!

**Review by joshbuchea — APPROVED** [review]

**Pull request merged**